sdk/src/types/block/core/mod.rs

```diff
@@ -136,10 +136,10 @@ impl Packable for Block {
     }
 }
 
-pub(crate) fn verify_parents(
-    strong_parents: &StrongParents,
-    weak_parents: &WeakParents,
-    shallow_like_parents: &ShallowLikeParents,
+pub(crate) fn verify_parents<const MAX: u8>(
+    strong_parents: &Parents<1, MAX>,
+    weak_parents: &Parents<0, MAX>,
+    shallow_like_parents: &Parents<0, MAX>,
```

Comment:
This doesn't seem very flexible, same MAX for all, no variable MIN. But anyway, for this function, lengths don't matter, maybe we can make it operate on slices and not need generics at all.